### PR TITLE
Convert userId to string before sending

### PR DESCRIPTION
**Pull request by Shivam4415**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/posthog/transform.js [hunk 1]
```diff
@@ -74,6 +74,14 @@ const responseBuilderSimple = (message, category, destination) => {
     ...payload.properties
   };
 
+  // Convert the distinct_id to string as that is the needed type in destinations.
+  if (payload.distinct_id) {
```
Comment (arnab-p) [inline v0/destinations/posthog/transform.js:78]: This will fail for falsy values. Let's check for not null and not undefined.

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: v0/destinations/posthog/transform.js (hunk 1 above)
Comment (Shivam4415) [reply v0/destinations/posthog/transform.js:78]: Javascript supports 7 falsy values :: 
1. the number 0, -0
2. the BigInt 0n
3. the keyword null
4. the keyword undefined
5. the boolean false
6. the number NaN
7. the empty string "" (equivalent to '' or ``)
I tested for all of them and it never fails.
Should I still change it the way u are suggesting.

Comment (arnab-p) [reply v0/destinations/posthog/transform.js:78]: It's improbable, but if `payload.distinct_id` is `0`, it won't be converted to String. Can we make sure that we're handling all the cases?

Comment (Shivam4415) [reply v0/destinations/posthog/transform.js:78]: Actually what I was thinking is  if payload.distinct_id is 0, which should never happen actually but will happen only if wrong payload is sent or sdk failed to generate correct anonymousId/userId. Keeping that in my mind I thought why I should handle logic for wrong value. Anyways I will add logic as u suggested and commit.

File: v0/destinations/posthog/transform.js [hunk 2]
```diff
@@ -74,6 +74,21 @@ const responseBuilderSimple = (message, category, destination) => {
     ...payload.properties
   };
 
+  // Convert the distinct_id to string as that is the needed type in destinations.
+  if (!(payload.distinct_id === null || payload.distinct_id === undefined)) {
```
Comment (arnab-p) [inline v0/destinations/posthog/transform.js:78]: Let's use the method `isDefinedAndNotNull` from https://github.com/rudderlabs/rudder-transformer/blob/738121e00997ebc438cc41318781f8a9b5ab53fb/v0/util/index.js#L30

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: v0/destinations/posthog/transform.js (hunk 2 above)
Comment (Shivam4415) [reply v0/destinations/posthog/transform.js]: fixed as suggested

**Review by arnab-p — APPROVED** [review]

**Pull request merged**